src/app/chip-zcl-zpro/command-encoder/decoder.c

```diff
@@ -103,18 +103,16 @@ void printApsFrame(EmberApsFrame * frame)
            frame->groupId, frame->sequence, frame->radius);
 }
 
-// Size of an encoded EmberApsFrame.
-static const size_t kEmberApsFrameSize = 13;
-
 uint16_t extractMessage(uint8_t * buffer, uint16_t buffer_length, uint8_t ** msg)
 {
     // The message starts after the EmberApsFrame.
     uint16_t result = 0;
-    if (msg && buffer_length > kEmberApsFrameSize)
+    EmberApsFrame frame;
+    uint16_t apsFrameSize = extractApsFrame(buffer, buffer_length, &frame);
+    if (msg && apsFrameSize > 0 && buffer_length > apsFrameSize)
```

Comment:
Why do we need the `buffer_length > apsFrameSize` check here?  We _might_ have a length-0 message; presumably trying to process it further will fail is all.  And if `extractApsFrame` is not lying to us, `buffer_length < apsFrameSize` can't happen.

Reply:
Yup this check is not needed anymore as this is now inside extractApsFrame